<table width="100%">
<tr>
<td width="50%" align="left"><a href="./02_RQ2_capacity_workforce.ipynb">← Previous: RQ2 — Capacity Workforce</a></td>
<td width="50%" align="right"><a href="./04_RQ4_healthcare_access.ipynb">Next: RQ4 — Healthcare Access →</a></td>
</tr>
</table>

## **3.0 Healthcare Expenditure and Financial Resources**

#### This section analyzes how healthcare expenditures have evolved across Canadian provinces and the patterns compare with population and demographic pressures.

> - #### **Research Question: How have healthcare expenditures evolved across Canadian provinces and territories, and how do expenditure patterns compare with population and demographic pressures?**

In [1]:
import pandas as pd
import numpy as np

from sqlalchemy import create_engine
from dotenv import load_dotenv

import os

import matplotlib.pyplot as plt

In [2]:
import pandas as pd
from daytascape_db_core.connection import get_db_engine

In [3]:
DATABASE = "health_system_performance"

engine = get_db_engine(DATABASE)

print("Connected database:", engine.url.database)

Connected database: health_system_performance


In [4]:
query = """
SELECT *
FROM master.analytics_province_year;
"""

df = pd.read_sql(query, engine)

print("Connected successfully to database!")

df.head(10)

Connected successfully to database!


,province_id,province_code,province_name,data_year,population,population_65_plus,population_80_plus,population_85_plus,population_65_share_pct,population_80_share_pct,...,private_health_expenditure_constant_2010_per_capita,provincial_government_health_expenditure_current,provincial_government_health_expenditure_per_capita,provincial_government_health_expenditure_constant_2010,provincial_government_health_expenditure_constant_2010_per_capi,territorial_government_health_expenditure_current,territorial_government_health_expenditure_per_capita,territorial_government_health_expenditure_constant_2010,territorial_government_health_expenditure_constant_2010_per_cap,nurses_total
0,15,PE,Prince Edward Island,2024,179709.0,36849.0,8218.0,3886.0,20.504816,4.572948,...,1914.110532,1.228779e+09,6881.989110,8.254174e+08,4622.892113,NaN,NaN,NaN,NaN,NaN
1,6,AB,Alberta,1975,1808689.0,134156.0,27442.0,12282.0,7.417306,1.517232,...,663.549287,6.949125e+08,384.207843,3.428803e+09,1895.739423,NaN,NaN,NaN,NaN,NaN
2,7,BC,British Columbia,1975,2499564.0,236960.0,51071.0,23239.0,9.480053,2.043196,...,679.794364,9.281825e+08,371.337761,4.379023e+09,1751.914626,NaN,NaN,NaN,NaN,NaN
3,7,BC,British Columbia,2022,5358845.0,1054174.0,250939.0,124925.0,19.671664,4.682707,...,2273.051979,3.010337e+10,5618.936230,2.317900e+10,4326.469070,NaN,NaN,NaN,NaN,108708.0
4,9,NB,New Brunswick,2021,790802.0,178652.0,39837.0,19193.0,22.591243,5.037544,...,2191.698134,3.945472e+09,4989.203072,3.151362e+09,3985.020465,NaN,NaN,NaN,NaN,NaN
5,5,CA,Canada,1975,23143275.0,1957075.0,377982.0,160362.0,8.456344,1.633226,...,568.363206,8.709289e+09,376.320519,4.111742e+10,1776.646681,NaN,NaN,NaN,NaN,NaN
6,8,MB,Manitoba,1975,1024975.0,104215.0,22287.0,9920.0,10.167565,2.174394,...,494.535967,3.766992e+08,367.520427,1.852127e+09,1806.997401,NaN,NaN,NaN,NaN,NaN
7,14,ON,Ontario,1980,8746013.0,850356.0,165961.0,70416.0,9.722785,1.897562,...,776.538159,5.164564e+09,590.505011,1.629120e+10,1862.700006,NaN,NaN,NaN,NaN,NaN
8,16,QC,Quebec,2020,8551095.0,1673371.0,401828.0,204785.0,19.569084,4.699141,...,1755.558556,4.704125e+10,5501.196296,3.642464e+10,4259.646558,NaN,NaN,NaN,NaN,NaN
9,14,ON,Ontario,2016,13876500.0,2260652.0,593482.0,302656.0,16.291226,4.276885,...,2000.807682,5.655294e+10,4075.446981,5.104966e+10,3678.856840,NaN,NaN,NaN,NaN,NaN


In [5]:
query = """
SELECT
    province_code,
    province_name,
    data_year,
    total_health_expenditure_current,
    total_health_expenditure_per_capita,
    total_health_expenditure_constant_2010,
    total_health_expenditure_constant_2010_per_capita,
    public_health_expenditure_current,
    public_health_expenditure_per_capita,
    public_health_expenditure_constant_2010,
    public_health_expenditure_constant_2010_per_capita,
    private_health_expenditure_current,
    private_health_expenditure_per_capita,
    private_health_expenditure_constant_2010,
    private_health_expenditure_constant_2010_per_capita,
    provincial_government_health_expenditure_current,
    provincial_government_health_expenditure_per_capita,
    provincial_government_health_expenditure_constant_2010,
    provincial_government_health_expenditure_constant_2010_per_capi,   territorial_government_health_expenditure_current,
    territorial_government_health_expenditure_per_capita,
    territorial_government_health_expenditure_constant_2010
    FROM master.analytics_province_year
WHERE province_code NOT IN ('CA', 'NT', 'YT', 'NU')
ORDER BY province_name, data_year;
"""

df_expenditure = pd.read_sql(query, engine)

df_expenditure

,province_code,province_name,data_year,total_health_expenditure_current,total_health_expenditure_per_capita,total_health_expenditure_constant_2010,total_health_expenditure_constant_2010_per_capita,public_health_expenditure_current,public_health_expenditure_per_capita,public_health_expenditure_constant_2010,...,private_health_expenditure_per_capita,private_health_expenditure_constant_2010,private_health_expenditure_constant_2010_per_capita,provincial_government_health_expenditure_current,provincial_government_health_expenditure_per_capita,provincial_government_health_expenditure_constant_2010,provincial_government_health_expenditure_constant_2010_per_capi,territorial_government_health_expenditure_current,territorial_government_health_expenditure_per_capita,territorial_government_health_expenditure_constant_2010
0,AB,Alberta,1971,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,None,None,None
1,AB,Alberta,1972,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,None,None,None
2,AB,Alberta,1973,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,None,None,None
3,AB,Alberta,1974,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,None,None,None
4,AB,Alberta,1975,9.923433e+08,548.653375,4.939117e+09,2730.771733,7.577722e+08,418.962157,3.738962e+09,...,129.691218,1.200154e+09,663.549287,6.949125e+08,384.207843,3.428803e+09,1895.739423,None,None,None
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
545,SK,Saskatchewan,2021,1.070319e+10,9165.958661,8.509617e+09,7287.434135,8.030554e+09,6877.175979,6.136982e+09,...,2288.782682,2.372635e+09,2031.867969,6.551220e+09,5610.309440,5.006469e+09,4287.421547,None,None,None
546,SK,Saskatchewan,2022,1.094586e+10,9288.316913,8.438291e+09,7160.469199,8.121616e+09,6891.748606,5.997589e+09,...,2396.568307,2.440702e+09,2071.102846,6.737869e+09,5717.544214,4.975730e+09,4222.248780,None,None,None
547,SK,Saskatchewan,2023,1.178260e+10,9743.268518,8.788776e+09,7267.613804,8.697784e+09,7192.370909,6.235791e+09,...,2550.897610,2.552985e+09,2111.114097,7.294317e+09,6031.815421,5.229589e+09,4324.450844,None,None,None
548,SK,Saskatchewan,2024,1.220294e+10,9842.150434,8.780179e+09,7081.560195,8.980860e+09,7243.417229,6.185453e+09,...,2598.733204,2.594726e+09,2092.748852,7.560850e+09,6098.123632,5.207439e+09,4200.004968,None,None,None


### **3.1 Nominal Expenditure Analysis (1975-2025)**

In [6]:
df_expenditure = (
    df_expenditure
       .sort_values(["province_code", "data_year"])
    .reset_index(drop=True)
)

In [7]:
df_expenditure.head(10)

,province_code,province_name,data_year,total_health_expenditure_current,total_health_expenditure_per_capita,total_health_expenditure_constant_2010,total_health_expenditure_constant_2010_per_capita,public_health_expenditure_current,public_health_expenditure_per_capita,public_health_expenditure_constant_2010,...,private_health_expenditure_per_capita,private_health_expenditure_constant_2010,private_health_expenditure_constant_2010_per_capita,provincial_government_health_expenditure_current,provincial_government_health_expenditure_per_capita,provincial_government_health_expenditure_constant_2010,provincial_government_health_expenditure_constant_2010_per_capi,territorial_government_health_expenditure_current,territorial_government_health_expenditure_per_capita,territorial_government_health_expenditure_constant_2010
0,AB,Alberta,1971,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,None,None,None
1,AB,Alberta,1972,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,None,None,None
2,AB,Alberta,1973,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,None,None,None
3,AB,Alberta,1974,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,None,None,None
4,AB,Alberta,1975,9.923433e+08,548.653375,4.939117e+09,2730.771733,7.577722e+08,418.962157,3.738962e+09,...,129.691218,1.200154e+09,663.549287,6.949125e+08,384.207843,3.428803e+09,1895.739423,None,None,None
5,AB,Alberta,1976,1.158899e+09,619.968423,5.159845e+09,2760.327623,8.984619e+08,480.644183,3.946495e+09,...,139.324240,1.213350e+09,649.097579,8.115502e+08,434.149625,3.564735e+09,1907.002653,None,None,None
6,AB,Alberta,1977,1.272117e+09,652.949485,5.257678e+09,2698.649015,9.788628e+08,502.428489,3.973432e+09,...,150.520996,1.284246e+09,659.175001,8.790690e+08,451.206536,3.568345e+09,1831.552203,None,None,None
7,AB,Alberta,1978,1.465355e+09,724.619474,5.649946e+09,2793.903247,1.127658e+09,557.627953,4.278051e+09,...,166.991521,1.371894e+09,678.403055,1.020452e+09,504.614559,3.871339e+09,1914.380709,None,None,None
8,AB,Alberta,1979,1.780959e+09,849.302838,6.368744e+09,3037.123073,1.391877e+09,663.757693,4.846061e+09,...,185.545144,1.522683e+09,726.136163,1.258186e+09,600.003290,4.380593e+09,2089.014959,None,None,None
9,AB,Alberta,1980,2.153109e+09,982.693028,7.088447e+09,3235.213729,1.660055e+09,757.660090,5.271843e+09,...,225.032938,1.816604e+09,829.110159,1.510164e+09,689.248978,4.795834e+09,2188.850184,None,None,None


In [8]:
df_expenditure["expenditure_growth_pct"] = (
    df_expenditure
    .sort_values(["province_code", "data_year"])
    .groupby("province_code")["total_health_expenditure_current"]
    .pct_change(fill_method=None) * 100
)

In [9]:
df_norm_expenditure_change = (
    df_expenditure[
        df_expenditure["total_health_expenditure_current"].notna()
    ]
    .sort_values(["province_code", "data_year"])
    .groupby(["province_code", "province_name"])
    .agg(
        start_year=("data_year", "first"),
        end_year=("data_year", "last"),
        start_expenditure=("total_health_expenditure_current", "first"),
        end_expenditure=("total_health_expenditure_current", "last")
    )
    .reset_index()
)

In [10]:
df_norm_expenditure_change["expenditure_growth_pct"] =(
    (df_norm_expenditure_change["end_expenditure"]
    - df_norm_expenditure_change["start_expenditure"])
    / df_norm_expenditure_change["start_expenditure"]
) * 100

In [11]:
df_norm_expenditure_change

,province_code,province_name,start_year,end_year,start_expenditure,end_expenditure,expenditure_growth_pct
0,AB,Alberta,1975,2025,9.923433e+08,4.747609e+10,4684.240119
1,BC,British Columbia,1975,2025,1.383359e+09,5.959150e+10,4207.740339
2,MB,Manitoba,1975,2025,5.463341e+08,1.629860e+10,2883.266178
3,NB,New Brunswick,1975,2025,2.774799e+08,8.069881e+09,2808.275540
4,NL,Newfoundland and Labrador,1975,2025,2.644928e+08,6.014884e+09,2174.119790
5,NS,Nova Scotia,1975,2025,3.824727e+08,1.250899e+10,3170.558367
6,ON,Ontario,1975,2025,4.422820e+09,1.463556e+11,3209.102217
7,PE,Prince Edward Island,1975,2025,5.917024e+07,1.879851e+09,3077.020279
8,QC,Quebec,1975,2025,3.378773e+09,8.437682e+10,2397.262069
9,SK,Saskatchewan,1975,2025,4.411970e+08,1.271906e+10,2782.852982


In [12]:
df_norm_expenditure_change.sort_values("expenditure_growth_pct", ascending=False)

,province_code,province_name,start_year,end_year,start_expenditure,end_expenditure,expenditure_growth_pct
0,AB,Alberta,1975,2025,9.923433e+08,4.747609e+10,4684.240119
1,BC,British Columbia,1975,2025,1.383359e+09,5.959150e+10,4207.740339
6,ON,Ontario,1975,2025,4.422820e+09,1.463556e+11,3209.102217
5,NS,Nova Scotia,1975,2025,3.824727e+08,1.250899e+10,3170.558367
7,PE,Prince Edward Island,1975,2025,5.917024e+07,1.879851e+09,3077.020279
2,MB,Manitoba,1975,2025,5.463341e+08,1.629860e+10,2883.266178
3,NB,New Brunswick,1975,2025,2.774799e+08,8.069881e+09,2808.275540
9,SK,Saskatchewan,1975,2025,4.411970e+08,1.271906e+10,2782.852982
8,QC,Quebec,1975,2025,3.378773e+09,8.437682e+10,2397.262069
4,NL,Newfoundland and Labrador,1975,2025,2.644928e+08,6.014884e+09,2174.119790


### **3.2 Real Expenditure Growth (1975-2025)**

In [13]:
df_expenditure["real_expenditure_growth_pct"] = (
    df_expenditure
    .sort_values(["province_code", "data_year"])
    .groupby("province_code")["total_health_expenditure_constant_2010"]
    .pct_change(fill_method=None) * 100
)

In [14]:
df_real_expenditure_change = (
    df_expenditure[
        df_expenditure["total_health_expenditure_constant_2010"].notna()
    ]
    .sort_values(["province_code", "data_year"])
    .groupby(["province_code", "province_name"])
    .agg(
        start_year=("data_year", "first"),
        end_year=("data_year", "last"),
        start_real_expenditure=("total_health_expenditure_constant_2010", "first"),
        end_real_expenditure=("total_health_expenditure_constant_2010", "last")
    )
    .reset_index()
)

In [15]:
df_real_expenditure_change["real_expenditure_growth_pct"]= (
    (df_real_expenditure_change["end_real_expenditure"]
    - df_real_expenditure_change["start_real_expenditure"])
    / df_real_expenditure_change["start_real_expenditure"]
) * 100

In [16]:
df_real_expenditure_change.sort_values("real_expenditure_growth_pct", ascending=False)

,province_code,province_name,start_year,end_year,start_real_expenditure,end_real_expenditure,real_expenditure_growth_pct
0,AB,Alberta,1975,2025,4.939117e+09,3.546088e+10,617.959886
1,BC,British Columbia,1975,2025,6.391165e+09,4.239232e+10,563.295695
6,ON,Ontario,1975,2025,2.151520e+10,1.054109e+11,389.936795
5,NS,Nova Scotia,1975,2025,1.920107e+09,8.994341e+09,368.429269
7,PE,Prince Edward Island,1975,2025,2.824020e+08,1.282866e+09,354.269448
3,NB,New Brunswick,1975,2025,1.348867e+09,5.907456e+09,337.956803
2,MB,Manitoba,1975,2025,2.593401e+09,1.128865e+10,335.283740
4,NL,Newfoundland and Labrador,1975,2025,1.092319e+09,4.503713e+09,312.307453
9,SK,Saskatchewan,1975,2025,2.246947e+09,8.939243e+09,297.839602
8,QC,Quebec,1975,2025,1.452330e+10,5.664722e+10,290.043818


> ##### **Insights:** 
> - ##### Nominal healthcare expenditure increased across all Canadian provinces between 1975 and 2025, indicating impressive financial commitment to healthcare over the period. Real expenditure als0 increase, although it is less than norminal expensidure which is expected since the long-term nominal increase is a reflection of changes in prices rather than increases in the volume of healcare resources. The sustained growth in healthcare expenditure over the period also indicates that healthcare sector has required progressive funding.

In [17]:
df_norm_expenditure_change[
    [
        "province_code",
        "province_name",
        "start_year",
        "end_year",
        "start_expenditure",
        "end_expenditure",
        "expenditure_growth_pct"
    ]
].sort_values(
    "expenditure_growth_pct",
    ascending=False
)

,province_code,province_name,start_year,end_year,start_expenditure,end_expenditure,expenditure_growth_pct
0,AB,Alberta,1975,2025,9.923433e+08,4.747609e+10,4684.240119
1,BC,British Columbia,1975,2025,1.383359e+09,5.959150e+10,4207.740339
6,ON,Ontario,1975,2025,4.422820e+09,1.463556e+11,3209.102217
5,NS,Nova Scotia,1975,2025,3.824727e+08,1.250899e+10,3170.558367
7,PE,Prince Edward Island,1975,2025,5.917024e+07,1.879851e+09,3077.020279
2,MB,Manitoba,1975,2025,5.463341e+08,1.629860e+10,2883.266178
3,NB,New Brunswick,1975,2025,2.774799e+08,8.069881e+09,2808.275540
9,SK,Saskatchewan,1975,2025,4.411970e+08,1.271906e+10,2782.852982
8,QC,Quebec,1975,2025,3.378773e+09,8.437682e+10,2397.262069
4,NL,Newfoundland and Labrador,1975,2025,2.644928e+08,6.014884e+09,2174.119790


### **3.3 - Largest Expenditure Expansion and Contraction Across Provinces**

#### **3.3.1 - 10 Largest Norminal Expenditure Annual Changes** 

In [18]:
df_expenditure[
    [
        "province_code",
        "province_name",
        "data_year",
        "total_health_expenditure_current",
        "expenditure_growth_pct"
    ]
].dropna(
    subset=["expenditure_growth_pct"]
    ).sort_values("expenditure_growth_pct", ascending=False).head(10)

,province_code,province_name,data_year,total_health_expenditure_current,expenditure_growth_pct
11,AB,Alberta,1982,3.351650e+09,27.744974
394,PE,Prince Edward Island,1980,1.223141e+08,26.641776
64,BC,British Columbia,1980,2.879995e+09,25.842389
10,AB,Alberta,1981,2.623704e+09,21.856551
489,QC,Quebec,2020,7.038676e+10,21.609694
8,AB,Alberta,1979,1.780959e+09,21.537711
393,PE,Prince Edward Island,1979,9.658271e+07,21.510265
175,NB,New Brunswick,1981,6.811856e+08,21.105872
9,AB,Alberta,1980,2.153109e+09,20.896029
176,NB,New Brunswick,1982,8.193082e+08,20.276793


#### **3.3.2 - 10 Smallest Norminal Expenditure Annual Changes**

In [19]:
df_expenditure[
    [
        "province_code",
        "province_name",
        "data_year",
        "total_health_expenditure_current",
        "expenditure_growth_pct"
    ]
].dropna(
    subset=["expenditure_growth_pct"]
    ).sort_values("expenditure_growth_pct").head(10)

,province_code,province_name,data_year,total_health_expenditure_current,expenditure_growth_pct
24,AB,Alberta,1995,6.109972e+09,-3.544042
23,AB,Alberta,1994,6.334468e+09,-3.061069
233,NL,Newfoundland and Labrador,1984,7.517200e+08,-2.845034
517,SK,Saskatchewan,1993,2.301955e+09,-1.413106
245,NL,Newfoundland and Labrador,1996,1.253379e+09,-1.268369
207,NB,New Brunswick,2013,4.720447e+09,-1.111725
465,QC,Quebec,1996,1.734475e+10,-0.994246
418,PE,Prince Edward Island,2004,5.334607e+08,-0.769012
491,QC,Quebec,2022,7.591030e+10,-0.729508
297,NS,Nova Scotia,1993,2.025088e+09,-0.320404


> ##### **Insights:**
> - ##### The largest year-over-year increases in nominal healthcare expenditure were concentrated in the late 1970s and early 1980s, with Alberta recording the highest increase of 27.7% in 1982, while Quebec recorded a notable 21.6% increase in 2020. These periods indicate episodes of rapid expansion in nominal healthcare spending, although the nominal measure incorporates both changes in prices and changes in real healthcare resources.
> - ##### The largest annual declines were generally modest, with Alberta recording the largest decrease at 3.5% in 1995, followed by a 3.1% decline in 1994. These contractions indicate periods in which nominal healthcare expenditure temporarily decreased rather than following the broader long-term upward trend.


#### **3.3.3 10 Largest Real Expenditure Expansion and Contraction Across Provinces**

##### **3.3.4 - 10 Largest Real Expenditure Annual Changes**

In [20]:
df_expenditure[
    [
        "province_code",
        "province_name",
        "data_year",
        "total_health_expenditure_constant_2010",
        "real_expenditure_growth_pct"
    ]
].dropna(
    subset=["real_expenditure_growth_pct"]
).sort_values(
    "real_expenditure_growth_pct", ascending=False
).head(10)

,province_code,province_name,data_year,total_health_expenditure_constant_2010,real_expenditure_growth_pct
415,PE,Prince Edward Island,2001,5.978740e+08,17.612184
489,QC,Quebec,2020,5.651770e+10,16.542733
394,PE,Prince Edward Island,1980,3.827795e+08,16.011596
64,BC,British Columbia,1980,8.562468e+09,13.939302
393,PE,Prince Edward Island,1979,3.299494e+08,12.862958
8,AB,Alberta,1979,6.368744e+09,12.722213
11,AB,Alberta,1982,8.742502e+09,12.012383
301,NS,Nova Scotia,1997,3.347781e+09,11.980859
28,AB,Alberta,1999,1.278545e+10,11.651641
9,AB,Alberta,1980,7.088447e+09,11.300553


##### **3.3.5 - 10 Smallest Annual Real Expenditure Changes**

In [21]:
df_expenditure[
    [
        "province_code",
        "province_name",
        "data_year",
        "total_health_expenditure_constant_2010",
        "real_expenditure_growth_pct"
    ]
].dropna(
    subset=["real_expenditure_growth_pct"]
).sort_values(
    "real_expenditure_growth_pct"
    ).head(10)

,province_code,province_name,data_year,total_health_expenditure_constant_2010,real_expenditure_growth_pct
233,NL,Newfoundland and Labrador,1984,1.458141e+09,-6.576835
491,QC,Quebec,2022,5.648921e+10,-5.488323
23,AB,Alberta,1994,1.010394e+10,-5.220022
511,SK,Saskatchewan,1987,3.637917e+09,-4.660287
390,PE,Prince Edward Island,1976,2.701785e+08,-4.328403
517,SK,Saskatchewan,1993,3.637417e+09,-3.943674
24,AB,Alberta,1995,9.715192e+09,-3.847508
190,NB,New Brunswick,1996,2.542061e+09,-2.793106
418,PE,Prince Edward Island,2004,6.211117e+08,-2.701645
298,NS,Nova Scotia,1994,2.988748e+09,-2.660239


> ##### **Insights:** 
> - ##### The largest year-over-year increases in real healthcare expenditure included Prince Edward Island at 17.6% in 2001, Quebec at 16.5% in 2020, and British Columbia at 13.9% in 1980. Because these figures are adjusted for price changes, they indicate periods of substantial expansion in healthcare spending in real terms, rather than increases attributable solely to rising prices.
> - ##### The largest real declines were observed in Newfoundland and Labrador at 6.6% in 1984, Quebec at 5.5% in 2022, and Alberta at 5.2% in 1994. These periods represent contractions in real healthcare expenditure, suggesting that the financial resources devoted to healthcare, measured in constant 2010 dollars, decreased relative to the preceding year.


### **3.4 Per-capita Health Expenditure (2025)**

In [22]:
df_expenditure[
    [
        "province_code",
        "province_name",
        "data_year",
        "total_health_expenditure_per_capita"
    ]
].dropna(
    subset=["total_health_expenditure_per_capita"]
).sort_values(
    ["province_code", "data_year"]
)

,province_code,province_name,data_year,total_health_expenditure_per_capita
4,AB,Alberta,1975,548.653375
5,AB,Alberta,1976,619.968423
6,AB,Alberta,1977,652.949485
7,AB,Alberta,1978,724.619474
8,AB,Alberta,1979,849.302838
...,...,...,...,...
545,SK,Saskatchewan,2021,9165.958661
546,SK,Saskatchewan,2022,9288.316913
547,SK,Saskatchewan,2023,9743.268518
548,SK,Saskatchewan,2024,9842.150434


In [23]:
df_expenditure[
    (df_expenditure["data_year"] == 2025)
][
    [
        "province_code",
        "province_name",
        "total_health_expenditure_per_capita"
    ]
].sort_values(
    "total_health_expenditure_per_capita", ascending=False)

,province_code,province_name,total_health_expenditure_per_capita
329,NS,Nova Scotia,11617.101486
274,NL,Newfoundland and Labrador,11108.912364
164,MB,Manitoba,10867.900293
439,PE,Prince Edward Island,10469.804929
109,BC,British Columbia,10438.662180
549,SK,Saskatchewan,10191.025206
54,AB,Alberta,9597.481968
219,NB,New Brunswick,9431.232922
494,QC,Quebec,9319.961549
384,ON,Ontario,9044.897734


In [24]:
df_expenditure["per_capita_expenditure_growth_pct"] = (
    df_expenditure
    .sort_values(["province_code", "data_year"])
    .groupby("province_code")["total_health_expenditure_per_capita"]
    .pct_change(fill_method=None) 
    * 100
)

#### **3.4.1 Largest Changes in Per Capita Health Expenditure**

In [25]:
df_expenditure[
    [
        "province_code",
        "province_name",
        "data_year",
        "total_health_expenditure_per_capita",
        "per_capita_expenditure_growth_pct"
    ]

].dropna(
    subset=["per_capita_expenditure_growth_pct"]
).sort_values(
    "per_capita_expenditure_growth_pct", ascending=False
).head(10)

,province_code,province_name,data_year,total_health_expenditure_per_capita,per_capita_expenditure_growth_pct
394,PE,Prince Edward Island,1980,988.516264,25.771808
11,AB,Alberta,1982,1414.301699,23.501429
64,BC,British Columbia,1980,1048.849548,22.147449
175,NB,New Brunswick,1981,964.253966,21.068328
489,QC,Quebec,2020,8231.315310,20.643924
393,PE,Prince Edward Island,1979,785.960131,20.322700
176,NB,New Brunswick,1982,1158.103246,20.103550
285,NS,Nova Scotia,1981,922.763828,19.716119
174,NB,New Brunswick,1980,796.454350,19.149892
120,MB,Manitoba,1981,1113.650320,18.589607


#### **3.4.2 Smallest Changes in Per Capita Health Expenditure**

In [26]:
df_expenditure[
    [
        "province_code",
        "province_name",
        "data_year",
        "total_health_expenditure_per_capita",
        "per_capita_expenditure_growth_pct"
    ]
].dropna(
    subset=["per_capita_expenditure_growth_pct"]
).sort_values(
    "per_capita_expenditure_growth_pct"
).head(10)

,province_code,province_name,data_year,total_health_expenditure_per_capita,per_capita_expenditure_growth_pct
24,AB,Alberta,1995,2234.386287,-4.740271
23,AB,Alberta,1994,2345.572804,-4.256883
233,NL,Newfoundland and Labrador,1984,1295.923737,-2.995942
491,QC,Quebec,2022,8752.299511,-1.887399
517,SK,Saskatchewan,1993,2286.179872,-1.697538
465,QC,Quebec,1996,2393.404220,-1.372378
381,ON,Ontario,2022,8160.517434,-1.189654
97,BC,British Columbia,2013,5741.085170,-1.152275
207,NB,New Brunswick,2013,6225.358705,-1.129983
418,PE,Prince Edward Island,2004,3874.585496,-1.094059


#### **3.4.3 Long-term per-capita Change**

In [27]:
df_per_capita_change = (
    df_expenditure[
        df_expenditure["total_health_expenditure_per_capita"].notna()
    ]
    .sort_values(["province_code", "data_year"])
    .groupby(["province_code", "province_name"])
    .agg(
    start_year=("data_year", "first"),
    end_year=("data_year", "last"),
    start_per_capita=("total_health_expenditure_per_capita", "first"),
    end_per_capita=("total_health_expenditure_per_capita", "last")
    )
    .reset_index()
)

In [28]:
df_per_capita_change["per_capita_change_pct"] = (
    (df_per_capita_change["end_per_capita"] - df_per_capita_change["start_per_capita"])
    / df_per_capita_change["start_per_capita"]
) * 100

In [29]:
df_per_capita_change.sort_values("per_capita_change_pct", ascending=False)

,province_code,province_name,start_year,end_year,start_per_capita,end_per_capita,per_capita_change_pct
5,NS,Nova Scotia,1975,2025,462.734448,11617.101486,2410.533103
4,NL,Newfoundland and Labrador,1975,2025,475.282523,11108.912364,2237.328183
3,NB,New Brunswick,1975,2025,409.862115,9431.232922,2201.074574
9,SK,Saskatchewan,1975,2025,480.913167,10191.025206,2019.098812
7,PE,Prince Edward Island,1975,2025,502.618333,10469.804929,1983.052735
2,MB,Manitoba,1975,2025,533.021897,10867.900293,1938.921921
1,BC,British Columbia,1975,2025,553.440035,10438.662180,1786.141501
0,AB,Alberta,1975,2025,548.653375,9597.481968,1649.279674
8,QC,Quebec,1975,2025,533.745865,9319.961549,1646.142154
6,ON,Ontario,1975,2025,531.602044,9044.897734,1601.441490


> ##### **Insights:** 
> - ##### The analyzed data reveals that per-capita healthcare expenditure increased substantially across all provinces between 1975 and 2025. Nova Scotia recorded the largest long term increase of 2,410.5%, followed by Newfoundland and Labrador (2,237.3%) nad New Brunswick (2,201%), while Ontario recorded the smallest increase amojng the provinces at 1,601.4%. In 2025, Nova Scotial also recorded the highest per-capita expenditure at $11,617, followed by Newfoundland and Labrador ($11,909) and Manitoba ($10,868), compared with Ontario ($9,045).
> - ##### Annual per-capita expenditure also showed considerable variations. The largest increases occured in Prince Ewdward Island 25.8% in 1980, Alberta 23.5% ub 1982, and British Columbia 22.1% in 1980. In contrast largest declines occured in Alberta -4.7% in 1995, Newfoundland and Labrador -3% in 1984, and Quebec -1.9% in 2022.
> - ##### Economically, the sustained increase in expenditure per person indicates that healthcare resources allocated relative to population expanded substantially over the period. However, differences in per-capita expenditue should be interpreted as differences in resource intensity rather than system performance, since higher spending per person does not by itself establish stronger access, capacity, or health outcomes.

### **3.5 Public vs. Private Expenditure Shares**

#### **3.5.1 Public vs Private Expenditure Comparison (2025)**

In [30]:
df_expenditure["public_share_pct"] = (
    df_expenditure["public_health_expenditure_current"]
    / df_expenditure["total_health_expenditure_current"]
) * 100

In [31]:
df_expenditure["private_share_pct"] = (
    df_expenditure["private_health_expenditure_current"]
    / df_expenditure["total_health_expenditure_current"]
) * 100

In [32]:
df_public_private_latest = (
    df_expenditure[
        df_expenditure["total_health_expenditure_current"].notna()
    ]
    .sort_values(["province_code", "data_year"])
    .groupby(["province_code", "province_name"])
    .tail(1)
    [
        [
            "province_code",
            "province_name",
            "data_year",
            "total_health_expenditure_current",
            "public_health_expenditure_current",
            "private_health_expenditure_current",
            "public_share_pct",
            "private_share_pct"
        ]
    ]
    .sort_values("public_share_pct", ascending=False)
    .reset_index(drop=True)
)
df_public_private_latest

,province_code,province_name,data_year,total_health_expenditure_current,public_health_expenditure_current,private_health_expenditure_current,public_share_pct,private_share_pct
0,MB,Manitoba,2025,1.629860e+10,1.281175e+10,3.486854e+09,78.606421,21.393579
1,NL,Newfoundland and Labrador,2025,6.014884e+09,4.695954e+09,1.318930e+09,78.072225,21.927775
2,PE,Prince Edward Island,2025,1.879851e+09,1.420861e+09,4.589900e+08,75.583696,24.416304
3,SK,Saskatchewan,2025,1.271906e+10,9.357803e+09,3.361256e+09,73.573075,26.426925
4,QC,Quebec,2025,8.437682e+10,6.143568e+10,2.294113e+10,72.811095,27.188905
5,BC,British Columbia,2025,5.959150e+10,4.321607e+10,1.637544e+10,72.520519,27.479481
6,NS,Nova Scotia,2025,1.250899e+10,9.028303e+09,3.480690e+09,72.174499,27.825501
7,AB,Alberta,2025,4.747609e+10,3.293818e+10,1.453791e+10,69.378454,30.621546
8,NB,New Brunswick,2025,8.069881e+09,5.548454e+09,2.521427e+09,68.755093,31.244907
9,ON,Ontario,2025,1.463556e+11,1.003791e+11,4.597650e+10,68.585770,31.414230


#### **3.5.2 Long-term Public vs Private Expenditure Growth**

In [33]:
df_public_private = df_expenditure[
    [
        "province_code",
        "province_name",
        "data_year",
        "public_share_pct",
        "private_share_pct"
    ]
].dropna(
    subset=["public_share_pct", "private_share_pct"]
).sort_values(["province_code", "data_year"]

).reset_index(drop=True)

In [34]:
df_public_private_change = (
    df_public_private
    .groupby(["province_code", "province_name"])
    .agg(
        start_year=("data_year", "first"),
        end_year=("data_year", "last"),
        start_public_share=("public_share_pct", "first"),
        end_public_share=("public_share_pct", "last")
    )
    .reset_index()
)

In [35]:
df_public_private_change["public_share_change_pp"] = (
    df_public_private_change["end_public_share"]
    - df_public_private_change["start_public_share"]
)

In [36]:
df_public_private_change = (
    df_public_private_change
    .sort_values("public_share_change_pp", ascending=False)
)

df_public_private_change

,province_code,province_name,start_year,end_year,start_public_share,end_public_share,public_share_change_pp
7,PE,Prince Edward Island,1975,2025,72.979676,75.583696,2.604020
2,MB,Manitoba,1975,2025,77.676002,78.606421,0.930419
1,BC,British Columbia,1975,2025,71.891418,72.520519,0.629100
4,NL,Newfoundland and Labrador,1975,2025,77.560044,78.072225,0.512181
9,SK,Saskatchewan,1975,2025,74.475842,73.573075,-0.902767
8,QC,Quebec,1975,2025,78.822496,72.811095,-6.011401
5,NS,Nova Scotia,1975,2025,78.793179,72.174499,-6.618679
6,ON,Ontario,1975,2025,75.257264,68.585770,-6.671494
0,AB,Alberta,1975,2025,76.361903,69.378454,-6.983448
3,NB,New Brunswick,1975,2025,77.414518,68.755093,-8.659425


> ##### **Insights:** 
> - ##### Public expenditure remained the dominant source of healthcare financing across all provinces, but the public share changed unevenly between 1975 and 2025. Prince Edward Island recorded the largest increase in the public share, rising from 73.0% to 75.6% (+2.6 pp), followed by Manitoba (+0.9 pp), British Columbia (+0.6 pp), and Newfoundland and Labrador (+0.5 pp). In contrast, New Brunswick recorded the largest decline, with its public share falling from 77.4% to 68.8% (−8.7 pp), followed by Alberta (−7.0 pp), Ontario (−6.7 pp), Nova Scotia (−6.6 pp), and Quebec (−6.0 pp).
> - ##### Overall, the results indicate a shift toward a relatively larger private share of healthcare expenditure in most province over the period. Six of the ten provinces experienced a decline in the public share, while four experienced an increase. This suggests that the financing composition of healthcare changed over time, although the extent of the shift varied considerably across jurisdictions.
> - ##### From a public-finance perspective, these differences are relevant because the composition of healthcare financing indicates how financial responsibility is distributed between public and private sources. A declining public share does not by itself indicate weaker public healthcare provision or poorer system performance, but it provides an important basis for examining government expenditure, resource allocation, and financial sustainability in the subsequent analysis.


#### **3.5.3 Provincial Level Expenditure Growth**

In [37]:
df_gov_expenditure = df_expenditure[
    [
        "province_code",
        "province_name",
        "data_year",
        "provincial_government_health_expenditure_current",
        "provincial_government_health_expenditure_per_capita",
        "provincial_government_health_expenditure_constant_2010"       
    ]
].copy()

In [38]:
df_gov_expenditure["gov_realexp_growth_pct"] = (
    df_gov_expenditure
    .sort_values(["province_code", "data_year"])
    .groupby("province_code")["provincial_government_health_expenditure_constant_2010"]
    .pct_change(fill_method=None) * 100
)                             

##### **3.5.3.1 Top 10 increases**

In [39]:
df_gov_expenditure[
    [
        "province_code",
        "province_name",
        "data_year",
        "gov_realexp_growth_pct"
    ]
].dropna(
    subset=["gov_realexp_growth_pct"]
).sort_values("gov_realexp_growth_pct", ascending=False
).head(10)

,province_code,province_name,data_year,gov_realexp_growth_pct
415,PE,Prince Edward Island,2001,18.302386
489,QC,Quebec,2020,15.797589
301,NS,Nova Scotia,1997,15.193072
11,AB,Alberta,1982,14.785524
64,BC,British Columbia,1980,13.238853
8,AB,Alberta,1979,13.154471
10,AB,Alberta,1981,12.279497
28,AB,Alberta,1999,12.235009
138,MB,Manitoba,1999,11.868311
248,NL,Newfoundland and Labrador,1999,11.423336


##### **3.5.3.2 Top 10 declines**

In [40]:
df_gov_expenditure[
    [
        "province_code",
        "province_name",
        "data_year",
        "gov_realexp_growth_pct"
    ]
].dropna(
    subset=["gov_realexp_growth_pct"]
).sort_values("gov_realexp_growth_pct"
).head(10)

,province_code,province_name,data_year,gov_realexp_growth_pct
23,AB,Alberta,1994,-10.051690
517,SK,Saskatchewan,1993,-7.204749
491,QC,Quebec,2022,-5.802263
298,NS,Nova Scotia,1994,-5.645503
24,AB,Alberta,1995,-5.335049
418,PE,Prince Edward Island,2004,-4.295680
22,AB,Alberta,1993,-4.227639
297,NS,Nova Scotia,1993,-3.654792
427,PE,Prince Edward Island,2013,-3.626217
516,SK,Saskatchewan,1992,-3.554638


##### **3.5.3.3 Largest Government Expenditure in 2025**

In [41]:
df_gov_expenditure_2025 = df_gov_expenditure[
    df_gov_expenditure["data_year"] == 2025
][
    [
        "province_code",
        "province_name",
        "provincial_government_health_expenditure_per_capita"
    ]
].dropna(
    subset=["provincial_government_health_expenditure_per_capita"]
).sort_values("provincial_government_health_expenditure_per_capita", ascending=False)

df_gov_expenditure_2025

,province_code,province_name,provincial_government_health_expenditure_per_capita
274,NL,Newfoundland and Labrador,8253.426439
329,NS,Nova Scotia,7784.912090
439,PE,Prince Edward Island,7446.470018
164,MB,Manitoba,7242.035155
109,BC,British Columbia,7136.784043
549,SK,Saskatchewan,6284.602242
494,QC,Quebec,6277.100015
54,AB,Alberta,6134.494078
219,NB,New Brunswick,5934.464616
384,ON,Ontario,5750.180743


> ##### **Insights:**
> - ##### In 2025, Newfoundland and Labrador had the highest provincial government healthcare expenditure per capita at approximately $8,253, followed by Nova Scotia ($7,785) and Prince Edward Island ($7,446). Ontario had the lowest at approximately $5,750 per person, followed by New Brunswick ($5,934) and Alberta ($6,134).
> - ##### This indicates substantial differences in the level of government financial resources allocated to healthcare relative to population across provinces. The difference between Newfoundland and Labrador and Ontario is approximately $2,503 per person, meaning government healthcare expenditure per capita in Newfoundland and Labrador was about 44% higher than in Ontario.
> - ##### However, this should be interpreted as a difference in government resource intensity, not healthcare-system performance. Higher government expenditure per capita may reflect differences in population characteristics, geographic conditions, healthcare needs, service costs, or other structural factors. It does not by itself indicate that a province has better access, capacity, or outcomes.

### **3.6 How much has healthcare expenditure changed relative to population and demographic ageing?**

In [42]:
#### Recalled from Notebook 2 (chps_resources) (RQ2 - Population)

query = """
    SELECT
    province_id,
    province_code,
    province_name,
    data_year,
    population,
    population_65_plus,
    population_80_plus,
    population_85_plus,
    population_65_share_pct,
    population_80_share_pct,
    population_85_share_pct
FROM master.analytics_province_year
WHERE province_code NOT IN ('CA', 'NT', 'YT', 'NU')
ORDER BY province_name, data_year;
"""

df_population_demand = pd.read_sql(query, engine)

df_population_demand
df_population_demand["population_growth_pct"] = (
    df_population_demand
    .groupby('province_code')["population"]
    .pct_change() * 100
)

df_population_demand
df_population_demand[
    [
        "province_name",
        "province_code",
        "population",
        "population_growth_pct"
    ]
    
].head()
df_population_demand["population_growth_pct"].describe()
df_population_change = (
    df_population_demand
    .sort_values(["province_code", "data_year"])
    .groupby(["province_code", "province_name"])
    .agg(
        start_year=("data_year", "min"),
        end_year=("data_year", "max"),
        start_population=("population", "first"),
        end_population=("population", "last")
        
    )
    .reset_index()
)
df_population_change
df_population_change["absolute_change"] =(
    df_population_change["end_population"] - 
    df_population_change["start_population"]
)

df_population_change["population_change_pct"] = (
    df_population_change["absolute_change"]
    / df_population_change["start_population"]
) * 100

df_population_change

,province_code,province_name,start_year,end_year,start_population,end_population,absolute_change,population_change_pct
0,AB,Alberta,1971,2025,1665717.0,5029346.0,3363629.0,201.932801
1,BC,British Columbia,1971,2025,2240470.0,5697536.0,3457066.0,154.300928
2,MB,Manitoba,1971,2025,998876.0,1509702.0,510826.0,51.140081
3,NB,New Brunswick,1971,2025,642471.0,869682.0,227211.0,35.365176
4,NL,Newfoundland and Labrador,1971,2025,530854.0,549911.0,19057.0,3.589876
5,NS,Nova Scotia,1971,2025,797294.0,1093245.0,295951.0,37.119431
6,ON,Ontario,1971,2025,7849027.0,16258260.0,8409233.0,107.137267
7,PE,Prince Edward Island,1971,2025,112591.0,182657.0,70066.0,62.230551
8,QC,Quebec,1971,2025,6137305.0,9058297.0,2920992.0,47.594050
9,SK,Saskatchewan,1971,2025,932038.0,1266959.0,334921.0,35.934264


In [43]:
df_population_demand_1975 = df_population_demand[
    (df_population_demand["data_year"].isin([1975, 2025]) & 
    ~(df_population_demand["province_code"]).isin(["CA", "NU", "NT", "YT"]))
].copy()

In [44]:
df_population_demand_1975 = (
    df_population_demand_1975
    .pivot(
        index=["province_code", "province_name"],
        columns="data_year",
        values=["population", "population_65_plus"]
    )
    .reset_index()
)

In [45]:
df_population_demand_1975

province_code              province_name population              \
data_year                                                1975        2025   
0                    AB                    Alberta  1808689.0   5029346.0   
1                    BC           British Columbia  2499564.0   5697536.0   
2                    MB                   Manitoba  1024975.0   1509702.0   
3                    NB              New Brunswick   677008.0    869682.0   
4                    NL  Newfoundland and Labrador   556496.0    549911.0   
5                    NS                Nova Scotia   826549.0   1093245.0   
6                    ON                    Ontario  8319795.0  16258260.0   
7                    PE       Prince Edward Island   117724.0    182657.0   
8                    QC                     Quebec  6330303.0   9058297.0   
9                    SK               Saskatchewan   917415.0   1266959.0   

          population_65_plus             
data_year               1975       2025  
0                   134156.0   779999.0  
1                   236960.0  1167246.0  
2                   104215.0   260365.0  
3                    59979.0   202747.0  
4                    35431.0   138514.0  
5                    78864.0   246752.0  
6                   721306.0  3069697.0  
7                    13024.0    38105.0  
8                   471206.0  1964452.0  
9                   100213.0   225759.0

In [46]:
df_population_demand_1975.columns = [
    "province_code",
    "province_name",
    "population_1975",
    "population_2025",
    "population_65_1975",
    "population_65_2025"
]

In [47]:
df_population_demand_1975

,province_code,province_name,population_1975,population_2025,population_65_1975,population_65_2025
0,AB,Alberta,1808689.0,5029346.0,134156.0,779999.0
1,BC,British Columbia,2499564.0,5697536.0,236960.0,1167246.0
2,MB,Manitoba,1024975.0,1509702.0,104215.0,260365.0
3,NB,New Brunswick,677008.0,869682.0,59979.0,202747.0
4,NL,Newfoundland and Labrador,556496.0,549911.0,35431.0,138514.0
5,NS,Nova Scotia,826549.0,1093245.0,78864.0,246752.0
6,ON,Ontario,8319795.0,16258260.0,721306.0,3069697.0
7,PE,Prince Edward Island,117724.0,182657.0,13024.0,38105.0
8,QC,Quebec,6330303.0,9058297.0,471206.0,1964452.0
9,SK,Saskatchewan,917415.0,1266959.0,100213.0,225759.0


In [48]:
df_population_demand_1975["population_growth_pct"] = (
    (df_population_demand_1975["population_2025"]
    - df_population_demand_1975["population_1975"])
    / df_population_demand_1975["population_1975"]
) * 100

In [49]:
df_population_demand_1975["population_65_growth_pct"] = (
    (df_population_demand_1975["population_65_2025"]
    - df_population_demand_1975["population_65_1975"])
    / df_population_demand_1975["population_65_1975"]
) * 100

In [50]:
df_population_demand_1975

,province_code,province_name,population_1975,population_2025,population_65_1975,population_65_2025,population_growth_pct,population_65_growth_pct
0,AB,Alberta,1808689.0,5029346.0,134156.0,779999.0,178.065826,481.411938
1,BC,British Columbia,2499564.0,5697536.0,236960.0,1167246.0,127.941193,392.591999
2,MB,Manitoba,1024975.0,1509702.0,104215.0,260365.0,47.291592,149.834477
3,NB,New Brunswick,677008.0,869682.0,59979.0,202747.0,28.459634,238.029977
4,NL,Newfoundland and Labrador,556496.0,549911.0,35431.0,138514.0,-1.183297,290.940137
5,NS,Nova Scotia,826549.0,1093245.0,78864.0,246752.0,32.266206,212.882938
6,ON,Ontario,8319795.0,16258260.0,721306.0,3069697.0,95.416594,325.574860
7,PE,Prince Edward Island,117724.0,182657.0,13024.0,38105.0,55.156977,192.575246
8,QC,Quebec,6330303.0,9058297.0,471206.0,1964452.0,43.094209,316.898766
9,SK,Saskatchewan,917415.0,1266959.0,100213.0,225759.0,38.100968,125.279155


In [51]:
df_expenditure_demand = df_real_expenditure_change.merge(
    df_population_demand_1975[
        [
            "province_code",
            "population_growth_pct",
            "population_65_growth_pct"
        ]
    ],
    on="province_code",
    how="inner"
    )

In [52]:
df_expenditure_demand["expenditure_population_gap_pp"] = (
    df_expenditure_demand["real_expenditure_growth_pct"]
    - df_expenditure_demand["population_growth_pct"])

In [53]:
df_expenditure_demand["expenditure_ageing_gap_pp"] = (
    df_expenditure_demand["real_expenditure_growth_pct"]
    - df_expenditure_demand["population_65_growth_pct"]
)

In [54]:
df_expenditure_demand[
    [
        "province_code",
        "province_name",
        "real_expenditure_growth_pct",
        "population_growth_pct",
        "population_65_growth_pct",
        "expenditure_population_gap_pp",
        "expenditure_ageing_gap_pp"
    ]
].sort_values("expenditure_population_gap_pp", ascending=False)

,province_code,province_name,real_expenditure_growth_pct,population_growth_pct,population_65_growth_pct,expenditure_population_gap_pp,expenditure_ageing_gap_pp
0,AB,Alberta,617.959886,178.065826,481.411938,439.894061,136.547948
1,BC,British Columbia,563.295695,127.941193,392.591999,435.354502,170.703696
5,NS,Nova Scotia,368.429269,32.266206,212.882938,336.163064,155.546332
4,NL,Newfoundland and Labrador,312.307453,-1.183297,290.940137,313.490750,21.367316
3,NB,New Brunswick,337.956803,28.459634,238.029977,309.497169,99.926826
7,PE,Prince Edward Island,354.269448,55.156977,192.575246,299.112471,161.694203
6,ON,Ontario,389.936795,95.416594,325.574860,294.520202,64.361935
2,MB,Manitoba,335.283740,47.291592,149.834477,287.992147,185.449263
9,SK,Saskatchewan,297.839602,38.100968,125.279155,259.738634,172.560447
8,QC,Quebec,290.043818,43.094209,316.898766,246.949609,-26.854948


> ##### **Insights:** 
> - ##### Real healthcare expenditure increased substantially faster than population growth in all ten provinces between 1975 and 2025. The largest expenditure–population gap was recorded in Alberta, where real healthcare expenditure increased by 618.0%, compared with population growth of 178.1%, producing a gap of +439.9 percentage points. British Columbia recorded a similar pattern, with real expenditure increasing by 563.3% compared with population growth of 127.9%, resulting in a +435.4 percentage-point gap. Even the smallest gap, in Quebec, was +246.9 percentage points, reflecting real expenditure growth of 290.0% compared with population growth of 43.1%.
> - ##### The comparison with population ageing presents a more differentiated picture. In nine of the ten provinces, real healthcare expenditure grew faster than the population aged 65+. Manitoba recorded the largest expenditure–ageing gap at +185.4 percentage points, with real expenditure increasing by 335.3% compared with 149.8% growth in its 65+ population. British Columbia recorded a +170.7 percentage-point gap, while Saskatchewan recorded +172.6 percentage points. Quebec was the only province where ageing growth exceeded real expenditure growth: its 65+ population increased by 316.9%, compared with 290.0% real expenditure growth, producing a −26.9 percentage-point gap.
> - ##### From a healthcare-sustainability perspective, these results indicate that real healthcare expenditure expanded considerably relative to overall population growth across all provinces and generally also exceeded the growth of the older population. However, the magnitude of these expenditure–demand gaps should not be interpreted as evidence that resources were sufficient or that healthcare performance improved. They indicate the relative pace of financial resource expansion compared with demographic change and provide a basis for examining whether expenditure growth corresponds with changes in healthcare capacity, access, and outcomes.


### **3.7 How does expenditure relate to healthcare capacity?**

In [55]:
query = """
SELECT
    Province_id,
    province_code,
    province_name,
    data_year,
    population,
    population_65_plus,
    population_80_plus,
    population_85_plus,
    population_65_share_pct,
    population_80_share_pct,
    population_85_share_pct,
    hospital_beds_total,
    current_hospital_beds_total,
    physicians_total,
    registered_nurses,
    licensed_practical_nurses,
    nurse_practitioners,
    registered_psychiatric_nurses,
    icu_beds,
    long_term_care_beds,
    mental_health_addictions_beds,
    obstetrics_beds,
    other_acute_care_beds,
    pediatrics_beds,
    rated_capacity_beds,
    rehabilitation_beds
FROM master.analytics_province_year
WHERE province_code NOT IN ('CA', 'NT', 'YT', 'NU')
ORDER BY province_name, data_year;
"""

df_healthcare_capacity = pd.read_sql(query, engine)

#### Changes in Bed Capacity Across Provinces
df_bed_capacity_change = (df_healthcare_capacity[
  df_healthcare_capacity["hospital_beds_total"].notna()
].groupby(
    ["province_code", "province_name"]
).agg(
    start_year=("data_year", "min"),
    end_year=("data_year", "max"),
    start_beds=("hospital_beds_total", "first"),
    end_beds=("hospital_beds_total", "last")
).reset_index()
)

##### Absolute Change in Bed Capacity
df_bed_capacity_change["bed_change_absolute"] = (
    df_bed_capacity_change["end_beds"] - df_bed_capacity_change["start_beds"]
)


 #### Changes in Bed Capacity Across Provinces
df_bed_capacity_change["bed_change_pct"] = (
    df_bed_capacity_change["bed_change_absolute"]
    / df_bed_capacity_change["start_beds"]
) * 100

In [56]:
df_expenditure_capacity = (
    df_expenditure[
    (df_expenditure["data_year"].isin([2009, 2022]))
    ]
    .sort_values(["province_code", "data_year"])
    .groupby(["province_code", "province_name"])
    .agg(
    start_real_expenditure=("total_health_expenditure_constant_2010", "first"),
    end_real_expenditure=("total_health_expenditure_constant_2010", "last")
).reset_index()

)

df_expenditure_capacity

,province_code,province_name,start_real_expenditure,end_real_expenditure
0,AB,Alberta,2.234440e+10,3.306343e+10
1,BC,British Columbia,2.321332e+10,3.731919e+10
2,MB,Manitoba,7.423659e+09,9.655991e+09
3,NB,New Brunswick,4.333456e+09,5.487962e+09
4,NL,Newfoundland and Labrador,3.060379e+09,4.339699e+09
5,NS,Nova Scotia,5.398446e+09,7.804030e+09
6,ON,Ontario,7.250287e+10,9.926811e+10
7,PE,Prince Edward Island,7.677928e+08,1.091039e+09
8,QC,Quebec,3.885767e+10,5.648921e+10
9,SK,Saskatchewan,5.966176e+09,8.438291e+09


In [57]:
df_expenditure_capacity["real_expenditure_growth_pct"] = (
    (df_expenditure_capacity["end_real_expenditure"] - df_expenditure_capacity["start_real_expenditure"]) / df_expenditure_capacity["start_real_expenditure"]
)* 100

df_expenditure_capacity

,province_code,province_name,start_real_expenditure,end_real_expenditure,real_expenditure_growth_pct
0,AB,Alberta,2.234440e+10,3.306343e+10,47.971861
1,BC,British Columbia,2.321332e+10,3.731919e+10,60.766276
2,MB,Manitoba,7.423659e+09,9.655991e+09,30.070507
3,NB,New Brunswick,4.333456e+09,5.487962e+09,26.641673
4,NL,Newfoundland and Labrador,3.060379e+09,4.339699e+09,41.802678
5,NS,Nova Scotia,5.398446e+09,7.804030e+09,44.560684
6,ON,Ontario,7.250287e+10,9.926811e+10,36.916106
7,PE,Prince Edward Island,7.677928e+08,1.091039e+09,42.100740
8,QC,Quebec,3.885767e+10,5.648921e+10,45.374683
9,SK,Saskatchewan,5.966176e+09,8.438291e+09,41.435493


In [58]:
df_expenditure_resource = df_expenditure_capacity.merge(
    df_bed_capacity_change[
        [
            "province_code",
            "start_year",
            "end_year",
            "bed_change_pct"
        ]
    ],
    on="province_code",
    how="inner"
)

In [59]:
df_expenditure_resource

,province_code,province_name,start_real_expenditure,end_real_expenditure,real_expenditure_growth_pct,start_year,end_year,bed_change_pct
0,AB,Alberta,2.234440e+10,3.306343e+10,47.971861,2009,2022,1.485981
1,BC,British Columbia,2.321332e+10,3.731919e+10,60.766276,2009,2022,-18.301639
2,MB,Manitoba,7.423659e+09,9.655991e+09,30.070507,2009,2022,0.893504
3,NB,New Brunswick,4.333456e+09,5.487962e+09,26.641673,2009,2022,-8.497669
4,NL,Newfoundland and Labrador,3.060379e+09,4.339699e+09,41.802678,2009,2022,1.675978
5,NS,Nova Scotia,5.398446e+09,7.804030e+09,44.560684,2009,2022,2.725832
6,ON,Ontario,7.250287e+10,9.926811e+10,36.916106,2009,2022,10.303818
7,PE,Prince Edward Island,7.677928e+08,1.091039e+09,42.100740,2009,2022,-19.295302
8,QC,Quebec,3.885767e+10,5.648921e+10,45.374683,2011,2022,0.095887
9,SK,Saskatchewan,5.966176e+09,8.438291e+09,41.435493,2009,2022,-2.560495


In [60]:
df_expenditure_resource["expenditure_capacity_gap"] = (

    df_expenditure_resource["real_expenditure_growth_pct"] 
    - df_expenditure_resource["bed_change_pct"]
)

In [61]:
df_expenditure_resource.sort_values("expenditure_capacity_gap", ascending=False)

,province_code,province_name,start_real_expenditure,end_real_expenditure,real_expenditure_growth_pct,start_year,end_year,bed_change_pct,expenditure_capacity_gap
1,BC,British Columbia,2.321332e+10,3.731919e+10,60.766276,2009,2022,-18.301639,79.067915
7,PE,Prince Edward Island,7.677928e+08,1.091039e+09,42.100740,2009,2022,-19.295302,61.396042
0,AB,Alberta,2.234440e+10,3.306343e+10,47.971861,2009,2022,1.485981,46.485880
8,QC,Quebec,3.885767e+10,5.648921e+10,45.374683,2011,2022,0.095887,45.278796
9,SK,Saskatchewan,5.966176e+09,8.438291e+09,41.435493,2009,2022,-2.560495,43.995988
5,NS,Nova Scotia,5.398446e+09,7.804030e+09,44.560684,2009,2022,2.725832,41.834852
4,NL,Newfoundland and Labrador,3.060379e+09,4.339699e+09,41.802678,2009,2022,1.675978,40.126701
3,NB,New Brunswick,4.333456e+09,5.487962e+09,26.641673,2009,2022,-8.497669,35.139343
2,MB,Manitoba,7.423659e+09,9.655991e+09,30.070507,2009,2022,0.893504,29.177003
6,ON,Ontario,7.250287e+10,9.926811e+10,36.916106,2009,2022,10.303818,26.612288


> ##### **Insights:** 
> - ##### Real healtcare expenditure increased faster than hopsital bed capacity across all ten provinces between 2009 - 2022. British Columbia recorded the larget expenditure-capacity gap, with real healthcare expenditure increasing by 60.8% while hospital beds declined by 18.3%, producing a 77.1 pecentage-point gap. Prince Edward Island recorded the second largest gap at 61.4 percentage points, while Ontario recorded the smallest at 26.6 percentage points, with expenditure increasing by 36.9% compared with 10.3% growth in beds. This shows that growth in financial resources exceeded growth in hospital capacity. This is probably because additional healthcare expenditure could mean spending on other health related priorities like workforce and access, which are beyond hospital-bed capacity expansion.

> #### **Section Summary:** 
> - #### Canadian healthcare system have substantially increased financial and workforce resources overtime, but hospital infrastructure (hospital-beds) has generally expanded more slowly than population and demographic ageing.

<table width="100%">
<tr>
<td width="50%" align="left"><a href="./02_RQ2_capacity_workforce.ipynb">← Previous: RQ2 — Capacity Workforce</a></td>
<td width="50%" align="right"><a href="./04_RQ4_healthcare_access.ipynb">Next: RQ4 — Healthcare Access →</a></td>
</tr>
</table>